themes/doom-monokai-pro-theme.el

```diff
@@ -0,0 +1,111 @@
+;; doom-monokai-pro-theme.el --- based off of Monokai Pro -*- no-byte-compile: t; -*-
+(require 'doom-themes)
+
+(defgroup doom-monokai-pro-theme nil
+  "Options for doom-themes"
+  :group 'doom-themes)
+
+(defcustom doom-monokai-pro-padded-modeline doom-themes-padded-modeline
+  "If non-nil, adds a 4px padding to the mode-line.
+Can be an integer to determine the exact padding."
+  :group 'doom-monokai-pro-theme
+  :type '(choice integer boolean))
+
+(def-doom-theme doom-monokai-pro
+  "A theme based off of Monokai Pro"
+
+  ;; name        gui       256       16
+  ((bg         '("#222222" nil       nil          ))
+   (bg-alt     '("#2b2b2b" nil       nil          ))
+   (base0      '("#0d0d0d" "black"   "black"      ))
+   (base1      '("#1b1b1b" "#1b1b1b"              ))
+   (base2      '("#212122" "#1e1e1e"              ))
+   (base3      '("#292b2b" "#292929" "brightblack"))
+   (base4      '("#3f4040" "#3f3f3f" "brightblack"))
+   (base5      '("#5c5e5e" "#525252" "brightblack"))
+   (base6      '("#757878" "#6b6b6b" "brightblack"))
+   (base7      '("#969896" "#979797" "brightblack"))
+   (base8      '("#ffffff" "#ffffff" "white"      ))
+   (fg         '("#b2bbc2" "#b2bbc2" "white"))
+   (fg-alt     (doom-darken fg 0.4))
+
+   (red        '("#cc6666" "#cc6666" "red"))
+   (blue       '("#81a2be" "#88aabb" "blue"))
+   (grey       '("#7a8590" "#7a8590" "brightblack"))
+   (violet     '("#a8a0ec" "#a8a0ec" "violet"))
+   (light-blue '("#90dbe6" "#90dbe6" "lightblue"))
+   (green      '("#b4d982" "#b4d982" "green"))
+   (yellow     '("#fad778" "#fad778" "yellow"))
+   (orange     '("#ef9c73" "#ef9c73" "orange"))
+   (pink       '("#ed6c88" "#ed6c88" "pink"))
+
+   (dark-blue  '("#81a2be" "#88aabb" "blue"))
+   (teal       '("#81a2be" "#88aabb" "blue"))
+   (cyan       '("#81a2be" "#88aabb" "blue"))
+   (dark-cyan  '("#81a2be" "#88aabb" "blue"))
+   (magenta    '("#a8a0ec" "#a8a0ec" "violet"))
+
+   ;; face categories
+   (highlight      blue)
+   (vertical-bar   `("#161616" ,@base0))
+   (selection      `(,(car (doom-lighten bg 0.1)) ,@(cdr base4)))
+   (builtin        light-blue)
+   (comments       grey)
+   (doc-comments   (doom-lighten grey 0.14))
+   (constants      violet)
+   (functions      green)
+   (keywords       pink)
+   (methods        green)
+   (operators      pink)
+   (type           light-blue)
+   (strings        yellow)
+   (variables      orange)
```

Comment:
Variables should just be white. Function/method parameters are orange when declaring a function. But not variables itself.

Reply:
Sometimes it's a little bit more difficult given that themes sometimes vary between modes (languages) and Emacs doesn't provide a robust API for that matter. I'll find some time to review this properly a the weekend.

Reply:
That's true. However, at least for JS and PHP, Monokai Pro doesn't vary in regards to variables.
I think the main problem is that major modes don't support the depth of customization some themes require. I only have experience with `php-mode` in regards to Monokai Pro. For example the concatenation operator (`.`) can't be fontified with what `php-mode` offers at the moment. I've actually submitted two PRs ([594](https://github.com/emacs-php/php-mode/pull/594), [606](https://github.com/emacs-php/php-mode/pull/606)). The latter is being reviewed right now. With those two PRs merged, flexibility with how php syntax can be fontified will be greatly improved.

However I don't know what the state of other major modes is.